components/arm/client.go

```diff
@@ -143,7 +143,18 @@ func (c *client) GoToInputs(ctx context.Context, goal []referenceframe.Input) er
 }
 
 func (c *client) DoCommand(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
-	return generic.DoFromConnection(ctx, c.conn, c.name, cmd)
+	command, err := protoutils.StructToStructPb(cmd)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := c.client.DoCommand(ctx, &commonpb.DoCommandRequest{
+		Name:    c.name,
+		Command: command,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return resp.Result.AsMap(), nil
```

Comment:
we should be able to pull this function out for all components (a more genericized version of `generic.DoFromConnection`). `DoFromResourceClient` perhaps?